src/ifcconvert/ColladaSerializer.cpp

```diff
@@ -213,11 +213,27 @@ void ColladaSerializer::ColladaExporter::ColladaScene::add(
 		{ (double)posmatrix[2], (double)posmatrix[5], (double)posmatrix[8], (double)posmatrix[11] },
 		{ 0, 0, 0, 1 }
 	};
-	
-	/// @todo: TFK: Rather than applying this offset to all leafs (which might be undesirable) should this offset be applied to a node higher up in the hierarchy?
-    matrix_array[0][3] += serializer->settings().offset[0];
-    matrix_array[1][3] += serializer->settings().offset[1];
-    matrix_array[2][3] += serializer->settings().offset[2];
+
+	gp_Trsf pos_trsf;
+	pos_trsf.SetValues(matrix_array[0][0], matrix_array[0][1], matrix_array[0][2], matrix_array[0][3],
```

Comment:
The prototype of `SetValues()` depends on the OCCT version. See [1].

I wonder if it is necessary to reconstruct a `gp_Trsf` manually. Can't we pick up the `gp_Trsf` from `transformation_towrite->data()`?

[1] https://github.com/IfcOpenShell/IfcOpenShell/blob/8ce4bbdf6194649cbaca4dff2ec1f53342cf5d39/src/ifcgeom/IfcGeomShapes.cpp#L729